File: src/main/java/com/brainstormer/controller/HomeController.java
package com.brainstormer.controller;

import org.springframework.security.access.prepost.PreAuthorize;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
public class HomeController {

	@PreAuthorize("hasRole('USER')")
	@RequestMapping("/")
	public String home() {
		return "home";
	}

	@PreAuthorize("permitAll()")
	@RequestMapping("/remember")
	public String remember() {
		return "remember";
	}
}

File: src/main/kotlin/spring/starter/controller/HomeController.kt
package spring.starter.controller

import org.springframework.security.access.prepost.PreAuthorize
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.RestController

@RestController
class HomeController {

    @PreAuthorize("hasRole('USER')")
    @GetMapping("/")
    fun home() = "home"

    @PreAuthorize("permitAll()")
    @GetMapping("/remember")
    fun remember() = "remember"
}